examples/06-verif-manual/vm-007-plastic_compression_of_a_pipe_assembly.py

```diff
@@ -0,0 +1,581 @@
+mapdl.keyopt(3, 3, 2)
+
+# Print
+print(mapdl.etlist())
+
+
+###############################################################################
+# Define Material
+# ~~~~~~~~~~~~~~~
+# Set up the material properties.
+#
+# * Young Modulus of steel is: :math:`E_{\mathrm{s}} = 26875000 psi`,
+# * Yield strength of steel is: :math:`\sigma_{\mathrm{(yp)s}} = 86000 psi`,
+# * Young Modulus of aluminum is: :math:`E_{\mathrm{a}} = 11000000  psi`,
+# * Yield strength of aluminum is: :math:`\sigma_{\mathrm{(yp)a}} = 55000 psi`,
+# * Poisson's ratio is: :math:`\nu = 0.3`
+
+# Steel material model.
+# Define Young's moulus and Poisson ratio for Steel.
+mapdl.mp("EX", 1, 26.875E6)
+mapdl.mp("PRXY", 1, 0.3)
+
+# Define non-linear material properties for Steel.
+mapdl.tb("BKIN", 1, 1)
+mapdl.tbtemp(0)
+mapdl.tbdata(1, 86000, 0)
+
+# aluminum material model.
+# Define Young's moulus and Poisson ratio for Aluminum.
+mapdl.mp("EX", 2, 11E6)
+mapdl.mp("PRXY", 2, 0.3)
+
+# Define non-linear material properties for Aluminum.
+mapdl.tb("BKIN", 2, 1)
+mapdl.tbtemp(0)
+mapdl.tbdata(1, 55000, 0)
+
+# Print
+print(mapdl.mplist())
+
+
+###############################################################################
+# Plot Stress - Strain Curve
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Use Matplotlib library to represent material model curves of Steel and Aluminum.
+
+# Define stress - strain properties of the steel.
+steel = {"stress_s": [0, 86000, 86000, 86000],
+         "strain_s": [0, 0.032, 0.1, 0.2]}
+
+# Define yielding strength point of the steel on the curve.
+xp = steel["strain_s"][1]
+yp = steel["stress_s"][1]
+
+# Set up the settings of the steel curve.
+plt.plot(steel["strain_s"], steel["stress_s"],
+         label="1020 CR STEEL", linewidth=2,
+         color='steelblue', linestyle="-", marker="o")
+plt.plot(xp, yp, marker='o')
+
+# Annotation settings
+plt.annotate(r'${(\sigma_{yp})_s}$', xy=(xp, yp), xytext=(0.05, 75000),
+             arrowprops=dict(facecolor='steelblue', shrink=0.05),
+             bbox=dict(facecolor='steelblue', edgecolor='black', boxstyle='round, pad=1'))
+
+# Define stress - strain properties of the aluminum.
+aluminum = {"stress_a": [0, 55000, 55000, 55000],
+            "strain_a": [0, 0.05, 0.1, 0.2]}
+
+# Define yielding strength point of the Aluminum on the curve.
+xp = aluminum["strain_a"][1]
+yp = aluminum["stress_a"][1]
+
+# Set up the settings of the aluminum curve.
+plt.plot(aluminum["strain_a"], aluminum["stress_a"],
+         label="2024-T4 Aluminum", linewidth=2,
+         color='sandybrown', linestyle="-", marker="o")
+plt.plot(xp, yp, marker='o')
+
+# Annotation settings
+plt.annotate(r'${(\sigma_{yp})_a}$', xy=(xp, yp), xytext=(0.07, 45000),
+             arrowprops=dict(facecolor='sandybrown', shrink=0.05),
+             bbox=dict(facecolor='sandybrown', edgecolor='black', boxstyle='round, pad=1'))
+
+plt.grid(True)
+plt.legend()
+plt.title("Stress - Strain Curve", fontsize=18)
+plt.show()
```

Comment:
If text location (``xytext``) is not an issue, this could be further reduced to:

```suggestion
# Define stress - strain properties of the steel.
steel = {"stress": [0, 86000, 86000, 86000],
         "strain": [0, 0.032, 0.1, 0.2]}

# Define stress - strain properties of the aluminum.
aluminum = {"stress": [0, 55000, 55000, 55000],
            "strain": [0, 0.05, 0.1, 0.2]}

def plot_yield_curve(material, label, color):
# Define yielding strength point of the steel on the curve.
    xp = material["strain"][1]
    yp = material["stress"][1]

    # Set up the settings of the steel curve.
    plt.plot(material["strain"], material["stress"],
            label=label, linewidth=2,
            color=color, linestyle="-", marker="o")
    plt.plot(xp, yp, marker='o')

    # Annotation settings
    plt.annotate(r'${(\sigma_{yp})}$', xy=(xp, yp),
                arrowprops=dict(facecolor=color, shrink=0.05),
                bbox=dict(facecolor=color, edgecolor='black', boxstyle='round, pad=1'))

plot_yield_curve(steel, "1020 CR STEEL", "steelblue")
plot_yield_curve(steel,"2024-T4 Aluminum", "sandybrown")

plt.grid(True)
plt.legend()
plt.title("Stress - Strain Curve", fontsize=18)
plt.show()
```

Reply:
I will leave it because it is useful and makes better the view of the graph.